plugins/org.eclipse.xtext/src/org/eclipse/xtext/formatting2/regionaccess/internal/LineRegion.java

```diff
@@ -36,6 +36,6 @@ public ILineRegion getNextLine() {
 
 	@Override
 	public ILineRegion getPreviousLine() {
-		return getTextRegionAccess().lineForOffset(getOffset() -1 );
+		return getTextRegionAccess().lineForOffset(getOffset() -2 );
```

Comment:
Hmm this looks really strange. Could you document why the prev line is actually line-2 ?